esptool.py

```diff
@@ -281,6 +281,53 @@ def sync(self):
         for i in range(7):
             self.command()
 
+    def _connect_attempt(self, mode='default_reset', legacy=False):
+        """ A single connection attempt, with legacy handling """
+        last_error = None
+        # issue reset-to-bootloader:
+        # RTS = either CH_PD/EN or nRESET (both active low = chip in reset
+        # DTR = GPIO0 (active low = boot to flasher)
+        #
+        # DTR & RTS are active low signals,
+        # ie True = pin @ 0V, False = pin @ VCC.
+        if mode != 'no_reset':
+            self._port.setDTR(False)  # IO0=HIGH
+            self._port.setRTS(True)   # EN=LOW, chip in reset
+            if mode == 'esp32r0' or legacy:
+                time.sleep(2.0)  # Need to sleep longer here!
```

Comment:
I will look into restoring that> As for the two seconds, even then it sometimes misses on the first pass (but rarely). I can try some different values and see what happens with that, though I don't believe it can be much less.